File: tournaments/isTournament/isTournament.py
def isTournament(n, fromV, toV):

    edges = []

    for i in range(n):
        line = []
        for j in range(n):
            line.append(False)
        edges.append(line)

    for i in range(len(fromV)):
        edges[ fromV[i] - 1 ][ toV[i] - 1 ] = True

    for i in range(n):
        for j in range(i + 1, n):
            if edges[i][j] == edges[j][i]:
                return False

    if len(fromV) != n * (n - 1) // 2:
        return False
    return True

File: tournaments/isTournament/isTournament.cpp
bool isTournament(int n, std::vector<int> fromV, std::vector<int> toV) {

  std::vector<bool> line(n, false);
  std::vector<std::vector<bool>> edges(n, line);

  for (int i = 0; i < fromV.size(); i++) {
    edges[ fromV[i] - 1 ][ toV[i] - 1 ] = true;
  }

  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      if (!(edges[i][j] ^ edges[j][i])) {
        return false;
      }
    }
  }

  if (fromV.size() != n * (n - 1) / 2) {
    return false;
  }
  return true;
}
